gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConfigureEndpointMinimumVersionsTask.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.conjure;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.gradle.conjure.api.EndpointMinimumVersion;
+import com.palantir.logsafe.Preconditions;
+import java.util.Set;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.java.archives.Manifest;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.bundling.Jar;
+
+public class ConfigureEndpointMinimumVersionsTask extends DefaultTask {
+    private final SetProperty<EndpointMinimumVersion> endpointVersions =
```

Comment:
Should we use a `List` here? I don't think we need to hash each value, not do we expect duplicates with the same verb/path and different versions.
Very supportive of the collection here rather than a `Map`, since it gives us more flexibility in the future.

Reply:
done